django/contrib/messages/storage/cookie.py

```diff
@@ -117,13 +123,13 @@ def stored_length(val):
         self._update_cookie(encoded_data, response)
         return unstored_messages
 
-    def _hash(self, value):
+    def _legacy_hash(self, value):
         """
+        # RemovedInDjango40Warning: pre-Django 3.1 hashes will be invalid.
         Create an HMAC/SHA1 hash based on the value and the project setting's
         SECRET_KEY, modified to make it unique for the present purpose.
         """
-        key_salt = 'django.contrib.messages'
-        return salted_hmac(key_salt, value).hexdigest()
```

Comment:
@felixxm Uf, now that we changed it I noticed that we might have a problem here. Shouldn't the `_legacy_hash` keep a hardcoded `key_salt = 'django.contrib.messages'` -- it never had a different salt…